srtcore/group.cpp

```diff
@@ -2300,27 +2328,6 @@ int CUDTGroup::recv(char* buf, int len, SRT_MSGCTRL& w_mc)
 
         int32_t next_seq = m_RcvBaseSeqNo;
 
-        // If this set is empty, it won't roll even once, therefore output
-        // will be surely empty. This will be checked then same way as when
-        // reading from every socket resulted in error.
-
-        vector<CUDTSocket*> ready_sockets;
-
-        for (CEPoll::fmap_t::const_iterator i = sready.begin(); i != sready.end(); ++i)
-        {
-            if (i->second & SRT_EPOLL_ERR)
-                continue; // broken already
-
-            if ((i->second & SRT_EPOLL_IN) == 0)
-                continue; // not ready for reading
-
-            // Check if this socket is in aheads
-            // If so, don't read from it, wait until the ahead is flushed.
-            SRTSOCKET   id = i->first;
-            CUDTSocket* ps = m_pGlobal->locateSocket_LOCKED(id);
-            if (ps)
-                ready_sockets.push_back(ps);
-        }
         leaveCS(CUDT::s_UDTUnited.m_GlobControlLock);
```

Comment:
This breaks the rule of enclosing the whole critical section code range within one function.

Reply:
Moved inside the `recv_CollectReadReady(..)` function.